Assets/Scripts/UI/PaletteSwap.cs

```diff
@@ -0,0 +1,64 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Netherlands3D.Twin
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using UnityEditor;
+    using UnityEngine;
+    using UnityEngine.UI;
+
+    public class PaletteSwap : MonoBehaviour
+    {
+        private Image image;
+        // [SerializeField] private List<Color> colors;
+        [SerializeField] private ColorPalette palette;
+        [SerializeField] private bool allowRuntimeUpdateChange;
+
+        private static readonly int PaletteTex = Shader.PropertyToID("_PaletteTex");
+
+        private void Awake()
+        {
+            UpdatePalette();
+        }
+
+        private void OnValidate()
+        {
+            image = GetComponent<Image>();
+        
+            UpdatePalette();
+        }
+
+        private void Update()
+        {
+            if (!allowRuntimeUpdateChange)
+                return;
+        
+            UpdatePalette();
```

Comment:
Hoe performance intensief is dit? Vooral als het aantal images op begint te lopen?

Reply:
dit is redelijk intensief, en dat willen we dus eigenlijk niet gebruiken. ik kan het ook weghalen